### PR TITLE
#172: Add pressio-linalg into romtools

### LINKED ISSUE #172: Add pressio-linalg into romtools
Move everything from [this repo](https://github.com/Pressio/pressio-linalg) into romtools

### PR DESCRIPTION
**Pull request by cwschilly**
Fixes #172

### CODE REVIEW

File: tests/romtools/test_vector_space_utils/test_orthogonalizer.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import numpy as np
 import scipy.sparse
 from romtools.vector_space.utils.orthogonalizer import *
-from pressiolinalg import test_utils
+from romtools.linalg.parallel_utils import *
```
Comment (fnrizzi) [inline tests/romtools/test_vector_space_utils/test_orthogonalizer.py]: can you pleae change this to only import what is used?

File: tests/romtools/test_vector_space_utils/test_scaler.py [hunk 2]
```diff
@@ -3,8 +3,8 @@
 import numpy as np
 import scipy.sparse
 from romtools.vector_space.utils.scaler import *
-import pressiolinalg.linalg as pla
-from pressiolinalg import test_utils
+import romtools.linalg.linalg as la
+from romtools.linalg.parallel_utils import *
```
Comment (fnrizzi) [inline tests/romtools/test_vector_space_utils/test_scaler.py]: same comment as above

File: tests/romtools/test_vector_space_utils/test_shifter.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import pytest
 import numpy as np
 from romtools.vector_space.utils.shifter import *
-from pressiolinalg import test_utils
+from romtools.linalg.parallel_utils import *
```
Comment (fnrizzi) [inline tests/romtools/test_vector_space_utils/test_shifter.py]: same

File: tests/romtools/test_vector_space_utils/test_truncater.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import pytest
 import numpy as np
 from romtools.vector_space.utils.truncater import *
-from pressiolinalg import test_utils
+from romtools.linalg.parallel_utils import *
```
Comment (fnrizzi) [inline tests/romtools/test_vector_space_utils/test_truncater.py]: same

File: tests/romtools/test_vector_space_utils/test_orthogonalizer.py (hunk 1 above)
Comment (cwschilly) [reply tests/romtools/test_vector_space_utils/test_orthogonalizer.py]: Fixed

File: tests/romtools/test_vector_space_utils/test_scaler.py (hunk 2 above)
Comment (cwschilly) [reply tests/romtools/test_vector_space_utils/test_scaler.py]: Fixed

File: tests/romtools/test_vector_space_utils/test_shifter.py (hunk 3 above)
Comment (cwschilly) [reply tests/romtools/test_vector_space_utils/test_shifter.py]: Fixed

File: tests/romtools/test_vector_space_utils/test_truncater.py (hunk 4 above)
Comment (cwschilly) [reply tests/romtools/test_vector_space_utils/test_truncater.py]: Fixed

**Review by fnrizzi — APPROVED** [review]

**Pull request merged**